scripts/Muon/GUI/Common/results_tab_widget/results_tab_view.py

```diff
@@ -0,0 +1,175 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import (absolute_import, division, print_function)
+
+from mantidqt.utils.qt import load_ui
+from qtpy import QtCore, QtWidgets
+
+from Muon.GUI.Common.list_selector.list_selector_presenter import ListSelectorPresenter
+from Muon.GUI.Common.list_selector.list_selector_view import ListSelectorView
+
+# Constants
+FIT_SELECTOR_COL0_WIDTH = 600
+# from_row, from_col, row_span, col_span
+FIT_SELECTOR_GRID_POS = (0, 1, 4, 1)
+LOG_SELECTOR_COL0_WIDTH = FIT_SELECTOR_COL0_WIDTH
+LOG_SELECTOR_GRID_POS = FIT_SELECTOR_GRID_POS
+
+ui_fitting_tab, _ = load_ui(__file__, "results_tab_view.ui")
+
+
+class ResultsTabView(QtWidgets.QWidget, ui_fitting_tab):
+    # Public signals
+    function_selection_changed = QtCore.Signal()
+    results_name_edited = QtCore.Signal()
+    output_results_requested = QtCore.Signal()
+
+    def __init__(self, parent=None):
+        super(ResultsTabView, self).__init__(parent)
+        self._init_layout()
+        self._init_signals()
+
+    def set_output_results_button_enabled(self, on):
+        """
+        Set the status of the output results button
+        :param on: If True then enable the button otherwise disable it
+        """
+        self.output_results_table_btn.setEnabled(on)
+
+    def results_table_name(self):
+        """Return the name of the output table."""
+        return self.results_name_editor.text()
+
+    def set_results_table_name(self, name):
+        """Set the name of the output table.
+
+        :param name: A new name for the table
+        """
+        self.results_name_editor.setText(name)
+
+    def set_fit_function_names(self, names):
+        """Set a new list of function names for the function selector. This blocks
+        signals from the widget during the update
+
+        :param names: A list of strings specifying function names used in known fits
+        """
+        function_selector = self.fit_function_selector
+        original_selection = function_selector.currentText()
+        function_selector.blockSignals(True)
+        function_selector.clear()
+        function_selector.addItems(names)
+        if original_selection:
```

Comment:
should this be: 
```suggestion
        if original_selection in names:
```

Reply:
up to here

Reply:
Yes that's clearer :+1: I'm not sure what Qt does actually if the text is not in the list so this is safer.

Reply:
I'll wait to push a fix until the code review is finished.